### ISSUE TITLE
xunit.runner.utility multiple assembly names breaks Nuget restore?

### ISSUE THREAD

**Issue by Sumo-MBryant** [xunit/xunit#1905]
The `xunit.runner.utility` Nuget package uses different assembly names for each targeted framework. As of `2.3.0` both `netstandard` and `netcoreapp` frameworks are targeted.

Consider the following project structure, `->` denotes a dependency:
* Class Library (`netstandard`) -> `xunit.runner.utility`
* Console Application (`netcoreapp`) -> Class Library

When you Nuget restore this, the Class Library selects the `xunit.runner.utility.netstandard15` assembly, *however* the Console Application selects the `xunit.runner.utility.netcoreapp10` assembly. Only the `xunit.runner.utility.netcoreapp10` assembly is referenced in the deps.json file.

Because the assembly names don't match, the Class Library complains when loaded that it cannot find `xunit.runner.utility.netstandard15`.

# Reproduction

1. Clone https://github.com/Sumo-MBryant/XUnitDeps
2. Execute `dotnet run -p Application`

# Expected Result

Application runs

# Actual Result

```
Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'xunit.runner.utility.netstandard15, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c'. The system cannot find the file specified.
   at XUnitDeps.Tests.Test()
   at Application.Program.Main(String[] args) in Application\Program.cs:line 7
```

# Workaround?

I don't know of a way to force Nuget to specify a particular framework for a package dependency, however building a new package of `xunit.runner.utility` where all targeted frameworks had the same name fixed the issue for me.

I also tried customising the `AssemblyLoadContext` to load the "correct" assembly, however .Net Core checks the simple name of the assembly and rejects such runtime manipulation.

Is unifying the assembly names the most sensible solution to this issue?

**Comment by bradwilson** [comment]
The only way we'd be able to fix this is to explode out the packages and force people to choose the one they want (f.e., `xunit.runner.utility.netstandard`).

The problem you're going to run into by trying to extend/consume it from a library is that each version has different capabilities. If you just try to target netstandard, you will be missing .NET Framework-related features, for example.

**Comment by Sumo-MBryant** [comment]
Ok thanks, I'll just change my class library to use `netcoreapp`.

**Comment by Konard** [comment]
I have encountered the same issue. In order to fix this, I have explicitly referenced `xunit.runner.utility.netstandard15.dll` library.

To do this, add to your application's project file these lines:
```XML
<ItemGroup>
  <PackageReference Include="xunit.runner.utility" Version="2.4.1" GeneratePathProperty="true" />
</ItemGroup>

<ItemGroup>
  <Reference Include="xunit.runner.utility.netstandard">
    <HintPath>$(Pkgxunit_runner_utility)\lib\netstandard1.5\xunit.runner.utility.netstandard15.dll</HintPath>
  </Reference>
</ItemGroup>
```

**Comment by markvincze** [comment]
Hey folks,

I have a similar issue in a different setup.  
I have a solution which has two projects:

 - One test project created with `dotnet new xunit`
 - One console project created with `dotnet new console`

Both projects are targeting `netcoreapp2.2`.  
In the console project I added a reference to the test project, and I also installed the `xunit.runner.utility` package.

In the console project I'm trying to programatically run the tests in the other project (with `AssemblyRunner`). But when I try to run the console project, I get an exception saying

```
System.IO.FileLoadException: 'Could not load file or assembly 'xunit.runner.utility.netcoreapp10, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c'. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)'
```

The same thing happens if I try in VS or with the `dotnet` CLI.

(I tried using the same trick @Konard posted (just changing the `netstandard`s to `netcoreapp` in the path), but still getting the same exception.)

Has anyone encountered this? Am I doing something wrong? (I can upload a full repro if needed)
